### ISSUE TITLE
ConcurrentModificationException when clearing obsolete job activation requests

### ISSUE THREAD

**Issue by koevskinikola** [camunda/zeebe#13867]
**Describe the bug**

A `ConcurrentModificationException` occurs when clearing obsolete job activation (long polling) requests. The exception is thrown when clearing the `activeRequests` LinkedList which isn't thread-safe ([code](https://github.com/camunda/zeebe/blob/8ef7c6ffd293ab11c9d2f2ee2f79f98b47941d1d/gateway/src/main/java/io/camunda/zeebe/gateway/impl/job/InFlightLongPollingActivateJobsRequestsState.java#L77)).

**To Reproduce**

**Expected behavior**

A `ConcurrentModificationException` doesn't occur. 

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
java.util.ConcurrentModificationException: null

at java.util.LinkedList$ListItr.checkForComodification
at java.util.LinkedList$ListItr.remove
at java.util.Collection.removeIf
at io.camunda.zeebe.gateway.impl.job.InFlightLongPollingActivateJobsRequestsState.removeObsoleteRequestsAndUpdateMetrics ( io/camunda.zeebe.gateway.impl.job/InFlightLongPollingActivateJobsRequestsState.java:77 )
at io.camunda.zeebe.gateway.impl.job.InFlightLongPollingActivateJobsRequestsState.getPendingRequests ( io/camunda.zeebe.gateway.impl.job/InFlightLongPollingActivateJobsRequestsState.java:71 )
at io.camunda.zeebe.gateway.impl.job.LongPollingActivateJobsHandler.resetFailedAttemptsAndHandlePendingRequests ( io/camunda.zeebe.gateway.impl.job/LongPollingActivateJobsHandler.java:245 )
at io.camunda.zeebe.gateway.impl.job.LongPollingActivateJobsHandler.lambda$onNotification$6 ( io/camunda.zeebe.gateway.impl.job/LongPollingActivateJobsHandler.java:181 )
at io.camunda.zeebe.scheduler.ActorJob.invoke ( io/camunda.zeebe.scheduler/ActorJob.java:92 )
at io.camunda.zeebe.scheduler.ActorJob.execute ( io/camunda.zeebe.scheduler/ActorJob.java:45 )
at io.camunda.zeebe.scheduler.ActorTask.execute ( io/camunda.zeebe.scheduler/ActorTask.java:119 )
at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask ( io/camunda.zeebe.scheduler/ActorThread.java:109 )
at io.camunda.zeebe.scheduler.ActorThread.doWork ( io/camunda.zeebe.scheduler/ActorThread.java:87 )
at io.camunda.zeebe.scheduler.ActorThread.run ( io/camunda.zeebe.scheduler/ActorThread.java:204 )
```

</p>
</details>

**Environment:**
- OS:  Linux
- Zeebe Version:  8.3.0-SNAPSHOT
- Configuration: 
- Error occured in CW32 benchmark ([src](https://console.cloud.google.com/logs/query;query=error_group%2528%22CKD6icXrpN_oLA%22%2529%0AlogName:%22stdout%22%0Aresource.type%3D%22k8s_container%22%0Aresource.labels.pod_name%3D%22medic-y-2023-cw-32-1f55f22-benchmark-zeebe-gateway-5fbb459ctcwr%22%0Aresource.labels.container_name%3D%22zeebe-gateway%22%0Aresource.labels.location%3D%22europe-west1-b%22%0Aresource.labels.project_id%3D%22zeebe-io%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Aresource.labels.namespace_name%3D%22medic-y-2023-cw-32-1f55f22-benchmark%22;cursorTimestamp=2023-08-10T20:21:27.752805343Z;startTime=2023-08-10T19:51:57.752Z;endTime=2023-08-10T20:51:57.752Z?project=zeebe-io)).

**Comment by npepinpe** [comment]
Yes, my bad, we have a state modification in a callback that gets executed on the gRPC executor and not the actor itself =/

**Comment by Zelldon** [comment]
> This was merged 11th of July though, and AFAIK not back ported, so I'm unsure how a trial cluster would have it. Can they use alpha versions?

Yes I think there was a decision to get them used alpha versions.

**Comment by npepinpe** [comment]
Fix: https://github.com/camunda/zeebe/pull/13875

Not really sure how to write a good regression test for that =/

**Comment by Zelldon** [comment]
Unfortunately, this is part of alpha4 https://console.cloud.google.com/errors/detail/CKaE2PeurPD7Ng;service=zeebe;time=P7D?project=camunda-cloud-240911. Luckily this is already fixed thanks to @npepinpe

**Comment by npepinpe** [comment]
Most likely, since we were modifying non-thread-safe structures concurrently :+1:
